src/components/App.js

```diff
@@ -1,17 +1,222 @@
-import React from 'react';
+import React from "react";
+import cx from "classnames"
+import Header from './Header';
+import Search from './Search';
+import SearchResults from './SearchResults';
+import FilmDetails from './FilmDetails';
+import FavouritesList from './FavouritesList';
 
 class App extends React.Component {
-  constructor(){
-    super();
-  }
-
-  render(){
-    return (
-      <div>
-        React cinema app
-      </div>
-    )
-  }
+    constructor() {
+        super();
+
+        this.state = {
+            query: "",
+            currentPage: 1,
+            hints: [],
+            films: [],
+            totalFilms: 0,
+            isClosed: true,
+            isMinified: false,
+            hasFired: false,
+            filmDetails: {},
+            showFavs: false,
+            favourites: [],
+        }
+
+        this.receiveTitleQuery = this.receiveTitleQuery.bind(this);
+        this.receiveFilmID = this.receiveFilmID.bind(this);
+        this.receivePageNum = this.receivePageNum.bind(this);
+        this.receiveMove = this.receiveMove.bind(this);
+        this.receiveFavListState = this.receiveFavListState.bind(this);
+        this.receiveFav = this.receiveFav.bind(this);
+        this.receiveSearchHint = this.receiveSearchHint.bind(this);
+        this.searchByTitle = this.searchByTitle.bind(this);
+        this.searchByID = this.searchByID.bind(this);
+        this.toggleVisible = this.toggleVisible.bind(this);
+        this.toggleMinify = this.toggleMinify.bind(this);
+    }
+
+    componentDidMount() {
+        if (window.localStorage.favourites != undefined) {
+            this.setState({
+                favourites: JSON.parse(window.localStorage.getItem("favourites"))
+            })
+        }
+    }
+
+    searchByTitle(query, pageNum = 1) {
+        fetch(`http://www.omdbapi.com/?apikey=507b4100&type=movie&s=${query}&page=${pageNum}`)
+            .then(response => response.json())
+            .then(body => {
+                this.setState({
+                    hints: [],
+                    films: body.Search,
+                    totalFilms: body.totalResults
+                });
+            })
+            .catch(error => {
+                alert(error);
+            });
+    }
+
+    searchByID(id) {
+        fetch(`http://www.omdbapi.com/?apikey=507b4100&type=movie&i=${id}`)
+            .then(response => response.json())
+            .then(body => {
+                this.setState({
+                    filmDetails: body,
+                    hints: []
+                }, () => this.toggleVisible())
+            })
+            .catch(error => {
+                alert(error);
+            })
+    }
+
+    receiveTitleQuery(query) {
+        this.setState({
+            query: query,
+            currentPage: 1,
+            filmDetails: {}
+        }, () => this.searchByTitle(query, this.state.currentPage));
```

Comment:
I would take the search function out of the callback and pass in a hardcoded page number. Since `searchByTitle` only uses params and not state inside it, it should be fine